tools/setup-envtest/README.md

```diff
@@ -0,0 +1,47 @@
+# Envtest Binaries Manager
+
+This is a small tool that manages binaries for envtest. It can be used to
+download new binaries, list currently installed and available ones, and
+clean up versions.
+
+To use it, just go-install it on 1.16+ (it's a separate, self-contained
+module):
+
+```shell
+go install sigs.k8s.io/controller-runtime/tools/setup-envtest@latest
+```
+
+For full documentation, run it with the `--help` flag, but here are some
+examples:
+
+```shell
+# download the latest envtest, and print out info about it
+setup-envtest fetch
+
+# download the latest 1.19 envtest, and print out the path
+setup-envtest fetch -k '1.19.+' -p path
+
+# switch to the most recent 1.21 envtest on disk
+source <(setup-envtest switch -k '1.21.*' -p env)
+
+# list all available local versions for darwin/amd64
+setup-envtest list-installed --os darwin --arch amd64
+
+# remove all 1.16 versions from disk
+setup-envtest cleanup -k '1.16.*'
+```
+
+## Where does it put all those binaries?
+
+By default, binaries are stored in an OS-specific cache directory, as per
+the OS's conventions (see Go's `os.UserCacheDir`).
+
+For example, on Linux this is `$XDG_CACHE_DIR`, or just `~/.config` if
+that's unset.
+
+There's an overall folder that holds all files, and inside that is
+a folder for each version/platform pair.  The exact directory structure is
+not guarnateed, except that the leaf directory will contain the names
+expected by envtest.  You should always use `setup-envtest fetch` or
+`setup-envtest switch` (generally with the `-p path` or `-p env` flags) to
+get the directory that you should use.
```

Comment:
Would not first be required to move forward with https://github.com/kubernetes-sigs/controller-runtime/issues/1234?

Is this one working if: 
- We remove the bins from: /usr/local/kubebuilder/bin
- Then, use it to install the binaries in the bin/install as we do in KB. 

Note that currently in KB side we can just run the tests with make test and not directly because of it. (https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/project-v3/Makefile#L56)

Reply:
This is an alternate approach to installing in GOBIN.  We could, for instance, follow up with something along those lines where we have a well-known symlink in the store directory, and have `use` or `switch` change that instead of having you fiddle with the environment variable, but I figured this was a decent first step that's backwards compatible (all CR versions support using `$KUBEBUILDER_ASSETS`)